packages/core/src/lib/utils.js

```diff
@@ -29,3 +29,46 @@ export function symbol() {
 export function isArray(value) {
     return Array.isArray(value);
 }
+
+/** @param {Array<() => void>} arr */
+export function run_all(arr) {
+	for (var i = 0; i < arr.length; i++) {
+		arr[i]();
+	}
+}
+
+/**
+ * @param {any} thing
+ * @returns {thing is Function}
+ */
+export function is_function(thing) {
+	return typeof thing === 'function';
+}
+
+export const noop = () => {};
+
+
+/**
+ * Fires the handler once immediately (unless corresponding arg is set to `false`),
+ * then listens to the given events until the render effect context is destroyed
+ * @param {EventTarget} target
+ * @param {Array<string>} events
+ * @param {(event?: Event) => void} handler
+ * @param {any} call_handler_immediately
+ */
+export function listen(target, events, handler, call_handler_immediately = true) {
+	if (call_handler_immediately) {
+		handler();
+	}
+
+	for (var name of events) {
+		target.addEventListener(name, handler);
```

Comment:
I recommend using Svelte's `on` method from `svelte/events` for this instead. Handles a few edge cases around event delegation which may or may not be relevant in this repo (I haven't checked)